### PR TITLE
Build fixes

### PR DESCRIPTION
**Pull request by ebd2**
2 build fixes. See commit messages.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -149,7 +159,8 @@ $(IMAGE_DIRS): %: %/Dockerfile check-links
 	@echo "Building [$*] image"
 	@echo
 	cd $(dir $<) && time $(SHELL) -c "( tar -czh . | docker build $(DBFLAGS_$@) -t $@ --label $(LABEL) - )"
-	@echo
+
+$(IMAGE_TESTS): test-%: % %/capabilities.txt
```
Comment (cawallin) [inline Makefile:184]: Maybe make release should still run the tests, separating out the image building is good though so you can test that separately.

Comment (ebd2) [reply Makefile:184]: fixed: release and snapshot now depend on all.

**Comment by ebd2** [comment]
@alandpost

File: Makefile [hunk 2]
```diff
@@ -133,6 +153,17 @@ $(FLAGDIR)/%.flags: %/Dockerfile $(FLAG_SH)
...
 
 #
+# Generate .test.rd files, adding a test target for an image to the dependants
+# of its corresponding image.
+#
+# Note that the .test.rd files need to be updated if the Makefile changes
+# because the Makefile is what generates the .test.rd files.
+#
+$(TEST_RDEPS): depends/%.test.rd: Makefile
+	-mkdir -p $(dir $@)
```
Comment (alandpost) [inline Makefile:163]: `$(@D)`

File: Makefile [hunk 3]
```diff
@@ -133,6 +153,17 @@ $(FLAGDIR)/%.flags: %/Dockerfile $(FLAG_SH)
...
+	echo $*.dependants: test-$* >"$@"
```
Comment (alandpost) [inline Makefile:164]: Should the foo.dependants targets be phony?

**Review by alandpost — APPROVED** [review]
Looks good to me, with a couple nits

File: Makefile (hunk 3 above)
Comment (ebd2) [reply Makefile:164]: Yup. Good catch. The TEST_RDEPS ones I can do in the Makefile; the image ones I'll have to do in the .d files.

File: Makefile (hunk 2 above)
Comment (ebd2) [reply Makefile:163]: https://github.com/Teradata/docker-images/pull/30#pullrequestreview-45551650

**Pull request merged**